### PR TITLE
Fixed Grey empty screen at Upload wizard

### LINKED ISSUE #5330: Grey empty screen at Upload wizard caption step after denying files permission

### PR DESCRIPTION
**Pull request by kanahia1**
Fixes #5330

What changes did you make and why?

➕ Added functionality to the cancel button of the dialog box, because people often don't see or read everything on the screen. Therefore, we must warn them with a toast message in case they click on the cancel button and also inform them that they need to accept permissions in order to access the upload facility.
🐛 Fixed grey empty screen at upload caption step
🐛 Fixed loop created by calling permissions `onResume`

**Tests performed (required)**

Tested betaDebug on Pixel 2 with API level 29.

**Screenshots (for UI changes only)**

https://github.com/commons-app/apps-android-commons/assets/114223204/ac558d34-d1f2-4d4a-8078-335ca774124d
https://github.com/commons-app/apps-android-commons/assets/114223204/14abe656-7c28-4d19-bedc-1f5203ba4835
https://github.com/commons-app/apps-android-commons/assets/114223204/54b03962-0056-41d0-bdd7-a81d6eb37a55

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 1]
```diff
@@ -143,6 +146,7 @@ public class UploadActivity extends BaseActivity implements UploadContract.View,
      */
     public static HashMap<Place,Boolean> nearbyPopupAnswers;
 
+    public boolean hasAllPermissions = true;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:149]: Please add javadoc explaining what it means exactly when true or when false.

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:149]: Better avoid public members. Would you mind making it private and creating a setter and getter?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
It works great, thanks! :-)

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -315,9 +315,7 @@
     <codeStyleSettings language="protobuf">
       <option name="RIGHT_MARGIN" value="80" />
       <indentOptions>
-        <option name="INDENT_SIZE" value="2" />
         <option name="CONTINUATION_INDENT_SIZE" value="2" />
-        <option name="TAB_SIZE" value="2" />
```
[the remark this answers is not in the document]
Comment (kanahia1) [reply .idea/codeStyles/Project.xml]: @nicolas-raoul, They were added by mistake and thanks for pointing out and I have added them commit.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 1 above)
Comment (kanahia1) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java]: Sure, I have added the suggested changes in newer commits.

**Comment by kanahia1** [comment]
@nicolas-raoul, I have made all the requested changes, Can you please review them

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 3]
```diff
@@ -224,6 +241,11 @@ public boolean isLoggedIn() {
         return sessionManager.isUserLoggedIn();
     }
 
+    @Override
+    protected void onStart() {
+        super.onStart();
+        checkStoragePermissions();
+    }
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:248]: Please add an empty line after each method.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 4]
```diff
@@ -252,13 +275,36 @@ protected void checkBlockStatus() {
...
+    public void checkStoragePermissions() {
+        // Check if all required permissions are granted
         final boolean hasAllPermissions = PermissionUtils.hasPermission(this, PERMISSIONS_STORAGE);
         if (hasAllPermissions) {
+            // All required permissions are granted, so enable UI elements and perform actions
             receiveSharedItems();
-        } else if (VERSION.SDK_INT >= VERSION_CODES.M) {
-            requestPermissions(PERMISSIONS_STORAGE, RequestCodes.STORAGE);
+            cvContainerTopCard.setVisibility(View.VISIBLE);
+        } else{
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:285]: 1 space before `{`

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 5]
```diff
@@ -700,6 +746,23 @@ protected void onDestroy() {
             uploadCategoriesFragment.setCallback(null);
         }
     }
+    /**
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:749]: 1 empty line between each method.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 6]
```diff
@@ -700,6 +746,23 @@ protected void onDestroy() {
...
+     * Get the value of the showPermissionDialog variable.
+     *
+     * @return {@code true} if Permission Dialog should be shown, {@code false} otherwise.
+     */
+    public boolean isShowPermissionsDialog() {
+        return showPermissionsDialog;
+    }
+    /**
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:757]: 1 empty line between each method.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I added a few other modifications requests, very minor. Thank you! :-)

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 6 above)
Comment (kanahia1) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:757]: Sure, I will try to complete all requested changes

**Comment by nicolas-raoul** [comment]
Merged, thanks a lot Kanahia for your pull request! :-)

**Pull request merged**